### PR TITLE
CUMULUS-2688: Update bulk operations to fetch granules by unique columns

### PR DESCRIPTION
**Pull request by npauzenga**
**Summary:** 

Addresses [CUMULUS-2688](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2688)

This is PR 2/3 for the above ticket. This PR address the changes to bulk operations. When a bulk operation operates on granules it now needs to accept and fetch granules by `granuleId` + `collectionId`, not just `granuleId`. This is because the unique identifiers changed in the Postgres switchover.

## Changes

* Updates `BULK_GRANULE`, `BULK_GRANULE_DELETE`, and `BULK_GRANULE_REINGEST` operations in bulk-operations to support collectionId
* Updates integration tests to use unique identifiers

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased Phase 3
 
+### Breaking Changes
```
Comment (npauzenga) [inline CHANGELOG.md:9]: Wondering if there should be migration instructions for this... 🤔.

File: packages/api/lambdas/bulk-operation.js [hunk 2]
```diff
@@ -5,38 +5,47 @@ const Logger = require('@cumulus/logger');
...
 const { updateGranuleStatusToQueued } = require('../lib/writeRecords/write-granules');
-const { getGranuleIdsForPayload } = require('../lib/granules');
+const { getGranulesForPayload } = require('../lib/granules');
 const { reingestGranule, applyWorkflow } = require('../lib/ingest');
 
 const log = new Logger({ sender: '@cumulus/bulk-operation' });
 
 async function applyWorkflowToGranules({
-  granuleIds,
+  granules,
```
[the remark this answers is not in the document]
Comment (nemreid) [reply packages/api/lambdas/bulk-operation.js:25]: Yeah, I think it should be something better across the stack, esp at the endpoint level if we're renaming there anyway. I just can't decide on a good name. 🤔

File: CHANGELOG.md (hunk 1 above)
Comment (nemreid) [reply CHANGELOG.md:9]: Definitely requires an update to https://github.com/nasa/cumulus-api/

**Review by nemreid — CHANGES_REQUESTED** [review]
All the changes look fine, but I'd really like to avoid overloading `granules` even more in this way. Let's discuss regarding another term when you're available.

File: CHANGELOG.md (hunk 1 above)
Comment (npauzenga) [reply CHANGELOG.md:9]: Yeah, good call. I'll update that repo once all three tickets for 2688 are in as they all have changes that'll need to be noted. It looks like we might need a new `rds-phase-3` branch there that gets released at the same time as the main cumulus phase 3 release too 🤔

**Review by nemreid — APPROVED** [review]

**Pull request merged**